src/lib.rs

```diff
@@ -124,6 +124,13 @@ impl XorName {
         XOR_NAME_BITS
     }
 
+    /// Returns a copy of `self`, with the `i`-th bit flipped.
+    pub fn with_flipped_bit(&self, index: usize) -> XorName {
+        let &XorName(mut bytes) = self;
+        bytes[index / 8] = bytes[index / 8] ^ (1 << (7 - index % 8));
```

Comment:
I'd consider this an arithmetic operation, and Rust's built-in arithmetic operations panic. E. g.
```
let n = 10;
1_u8 << n;
```
panics with "shift operation overflowed". If malicious code is able to trigger that in our code, I'd still consider our code to be at fault.

Alternatively, we could just define behaviour for other values, e. g. just return an unchanged clone of the name if `index >= XOR_NAME_BITS`.

But if we _do_ want to return a `Result` here: Do you know whether there's an appropriate error type in the standard library? Otherwise I guess I'll define a new `XorNameError`?